src/core/reducer.js

```diff
@@ -58,18 +88,17 @@ export function CreateGameReducer({ game, numPlayers, multiplayer }) {
     _initial: {},
   };
 
-  const events = new Events(game.flow, ctx.currentPlayer);
-  ctxWithAPI = events.attach(ctxWithAPI);
-
   const state = game.flow.init({ G: initial.G, ctx: ctxWithAPI });
 
-  const { ctx: ctxWithEvents } = events.update(state);
   initial.G = state.G;
   initial._undo = state._undo;
+
+  // TODO liked to use apiCtx.update() here also, albeit
+  // this code is more intricate than it looks.
+  const { ctx: ctxWithEvents } = apiCtx.events.update(state);
   initial.ctx = ctxWithEvents;
-  initial.ctx = random.update(initial.ctx);
-  initial.ctx = Random.detach(initial.ctx);
-  initial.ctx = Events.detach(initial.ctx);
+  initial.ctx = apiCtx.random.update(initial).ctx;
+  initial.ctx = apiCtx.detachFromContext(initial.ctx);
```

Comment:
```js
  state = apiCtx.update(state);
  initial.ctx = state.ctx;
  initial.ctx = apiCtx.detachFromContext(initial.ctx);
```

should work here.

A more interesting observation is given that `update` is always followed by a `detachFromContext`, shall we combine them into one call `detach` that accepts `state` and returns `state`?